portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -90,6 +90,36 @@
 	 */
 	public static final String[] EMPTY_ACTION_IDS = {null};
 
+	@Override
+	public void addResourcePermission(
+			long companyId, String name, int scope, long roleId,
+			String actionId, String[] groupIds)
+		throws Exception {
+
+		if (scope == ResourceConstants.SCOPE_COMPANY) {
+			addResourcePermission(
+				companyId, name, scope, String.valueOf(companyId), roleId,
+				actionId);
+		}
+		else if (scope == ResourceConstants.SCOPE_GROUP_TEMPLATE) {
+			addResourcePermission(
+				companyId, name, ResourceConstants.SCOPE_GROUP_TEMPLATE,
+				String.valueOf(GroupConstants.DEFAULT_PARENT_GROUP_ID), roleId,
+				actionId);
+		}
+		else if (scope == ResourceConstants.SCOPE_GROUP) {
+			for (String curGroupId : groupIds) {
+				removeResourcePermission(
```

Comment:
Thanks for renaming the method. I think it's much more clear now that it's an add operation... except that it's not always the case since in this line 112 it's also removing the action from any groupIds which are not included in the array parameter. 

I don't want to be very picky about this, since this is not so uncommon in our codebase to do this. But maybe setResourcePermissions() would be better and then setting is implemented as a simple add for some scopes and add+remove in others. What do you think?